src/pages/SubscriptionService.tsx

```diff
@@ -135,13 +206,62 @@ class SubscriptionService extends Component<Props, SubscriptionServiceState> {
     }).map(service => this.setState({ service }));
   };
 
+  public getHandleBlur = (prop: keyof ServiceMetadata | keyof Service) => (
+    event: FocusEvent<HTMLInputElement>
+  ) => {
+    const target = event.target;
+    const inputValue =
+      target.type === "checkbox" ? target.checked : target.value;
+    const value = inputValueMap(prop, inputValue);
+
+    checkValue(prop, value).fold(
+      () => this.handleError(prop),
+      () => this.setData(this.removeError(prop), prop, value)
+    );
+  };
+
+  public getHandleMetadataBlur = (
+    prop: keyof ServiceMetadata | keyof Service
+  ) => (event: FocusEvent<HTMLSelectElement | HTMLInputElement>) => {
+    const {
+      target: { name, value }
+    } = event;
+    const inputValue = inputValueMap(name, value);
+    value && value.length > 0
+      ? checkValue(prop, inputValue).fold(
+          () => this.handleError(name),
+          () => this.setMetadata(this.removeError(name), name, inputValue)
+        )
+      : this.setMetadata(this.removeError(name), name, inputValue);
+  };
+
   public handleSubmit = async () => {
-    const serviceDecoding = Service.decode(this.state.service);
-    if (serviceDecoding.isLeft()) {
+    const serviceToUpdate = {
+      ...this.state.service,
+      ...this.state.form,
+      service_metadata: {
+        ...(this.state.service && this.state.service.service_metadata),
+        ...this.state.form.metadata,
+        scope:
+          this.state.service && this.state.service.service_metadata
+            ? this.state.service.service_metadata.scope
+            : "LOCAL"
+      }
+    };
+
+    const serviceDecoding = Service.decode(serviceToUpdate);
+
+    if (
+      serviceDecoding.isLeft() ||
+      (!isContactExists(serviceToUpdate.service_metadata as ServiceMetadata) ||
+        !isMandatoryFieldsValid(serviceToUpdate as Service))
```

Comment:
Can we use instead the decoder defined in `io-functions-commons` [ValidService](https://github.com/pagopa/io-functions-commons/blob/master/src/models/service.ts#L182)?

Reply:
Hi @BurnedMarshal, you're absolutely right, we can use a decoder like [ValidService](https://github.com/pagopa/io-functions-commons/blob/master/src/models/service.ts#L182) as you mentioned instead of this utilities functions. Thanks.